my_game.py

```diff
@@ -173,12 +173,22 @@ def on_draw(self):
         # Clear screen so we can draw new stuff
         self.clear()
 
-        # Draw the the sprite list from the map if configured to be drawn
+        # Draw the sprite list from the map if configured to be drawn
         for layer_name, layer_sprites in self.tilemap.sprite_lists.items():
             if MAP_LAYER_CONFIG[layer_name].get("draw", True):
                 if MAP_LAYER_CONFIG[layer_name].get("line_of_sight", False):
-                    # FIXME: Add logic for drawing stuff that should only be drawn if players have line of sight here
-                    pass
+                    for s in layer_sprites:
+                        # Only if player has line of sight with tile, it can be drawn
+                        try:
+                            if arcade.has_line_of_sight(
+                                    point_1 = s.position,
+                                    point_2 = self.player.position,
+                                    walls = self.tilemap.sprite_lists["impassable"],
+                                    check_resolution = TILE_SIZE
+                            ):
+                                s.draw(pixelated=DRAW_PIXELATED)
+                        except ZeroDivisionError:
+                            pass
```

Comment:
If the distance is 0 (the error occurs), we should draw the sprite, not ignore it.